gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -170,17 +171,18 @@ public ChangeRequest reload(Dns.ChangeRequestOption... options) {
 
   /**
    * Returns {@code true} if the change request has been completed. The function makes an API call
-   * to Google Cloud DNS in order request the status update only if the status of the change request
-   * is {@link Status#PENDING}. If the status is already {@link Status#DONE}, the method returns
-   * {@code true} without an API call.
+   * to Google Cloud DNS in order to request the status update only if the status of the change
+   * request is {@link Status#PENDING}. If the status is already {@link Status#DONE}, the method
+   * returns {@code true} without an API call.
+   *
+   * @throws DnsException upon failure of the API call or if the associated zone was not found
    */
   public boolean isDone() {
     if (status() == Status.DONE) {
       return true;
     }
     ChangeRequest updated = reload(Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.STATUS));
-    return updated.status() == Status.DONE;
-
+    return updated == null || updated.status() == Status.DONE;
```

Comment:
We just extensively discussed this with @aozarov. The only way of getting null is having a zone, creating a change request, then deleting a zone, and calling reload. This means that the change request must have finished. If `isDone()` is used in a loop, it is safer to return `null`.

Related to this, I created #851 which stemmed of our discussion with @aozarov .